### PR TITLE
Combine should keep the type

### PR DESCRIPTION
**Pull request by WhiteBlackGoose**

### CODE REVIEW

**Comment by WhiteBlackGoose** [comment]
Due to the bug that combine didn't keep the type the tester for Plotly.NET fails even to run the tests... idk why, but fortunately with this fix it's all going to work

File: src/DynamicObj/ImmutableDynamicObj.fs [hunk 1]
```diff
@@ -100,10 +100,11 @@ type ImmutableDynamicObj internal (map : Map<string, obj>) =
             | :? 'a as o -> o |> Some
             | _ -> None
 
-    // Merges two DynamicObj (Warning: In case of duplicate property names the members of the second object override those of the first)
-    static member combine (targetObject : #ImmutableDynamicObj) (sourceObject : #ImmutableDynamicObj) =
-
-        let rec loop (targetObject : #ImmutableDynamicObj) (sourceObject : #ImmutableDynamicObj) (propsToCopy: (string*obj) list) (acc:ImmutableDynamicObj) =
+    // Merges two ImmutableDynamicObj (Warning: In case of duplicate property names the members of the second object override those of the first)
+    static member combine (targetObject : 'a :> ImmutableDynamicObj) (sourceObject : 'b :> ImmutableDynamicObj) =
```
Comment (baronfel) [inline src/DynamicObj/ImmutableDynamicObj.fs:104]: I'd change this to a `///`-comment, with the note in a `remarks` element, so that this potentially-relevant detail is available to users:

```fsharp
/// <summary>Merges two ImmutableDynamicObj</summary>
/// <remarks>Warning: In case of duplicate property names the members of the second object override those of the first.</remarks>
```

Comment (WhiteBlackGoose) [reply src/DynamicObj/ImmutableDynamicObj.fs:105]: Thanks! Added

File: src/DynamicObj/DynamicObj.fsproj [hunk 2]
```diff
@@ -32,6 +32,7 @@
   
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Update="FSharp.Core" Version="6.0.1" />
```
Comment (kMutagene) [inline src/DynamicObj/DynamicObj.fsproj:35]: Why is this needed?

Comment (WhiteBlackGoose) [reply src/DynamicObj/DynamicObj.fsproj:35]: There's a bug in the newest SDK because of which it doesn't build. I can remove it as soon as you approve the rest

File: src/DynamicObj/ImmutableDynamicObj.fs (hunk 1 above)
Comment (kMutagene) [reply src/DynamicObj/ImmutableDynamicObj.fs:105]: If we the properties of the second object replaces those of the first object, it might also be more consistent to keep the second objects type instead of the first one

Comment (kMutagene) [reply src/DynamicObj/ImmutableDynamicObj.fs:105]: which is also better for pipelines IMO as

```fsharp
'A
|> IDO.combine 'B
```
can be chained into functions that take type A as input:

```fsharp
'A
|> IDO.combine 'B
|> A.doSomething
```

**Comment by WhiteBlackGoose** [comment]
@kMutagene @muehlhaus could you please update the nuget package when you have time?
